api/system.cc

```diff
@@ -158,6 +162,27 @@ void set_system(http_context& ctx, routes& r) {
             return json::json_return_type(json::json_void());
         });
     });
+
+    hs::dump_profile.set(r, [](std::unique_ptr<request> req) {
+        if (!__llvm_profile_dump) {
+            apilog.info("Executable is not instrumented with profile dumping, skipping");
+            return make_ready_future<json::json_return_type>(json::json_return_type(json::json_void()));
+        }
+        sstring profile_dest(__llvm_profile_get_filename ? __llvm_profile_get_filename() : "disk");
```

Comment:
I'm confused, what is proefile_dest actually used  for? You don't pass it to llvm_profile_dump... So is it just for printing? In this case, what is that "disk" case and when can it happen?

Reply:
It is only for printing. Basically if the files are instrumented both functions should be linked in the executable.
However, since I tried to implemented this API in the most forward compatible way possible, I didn't want to assume that this will always be the case (as we update to newer llvm versions). OTOH is will be kind of irrational
to fail the API call just because we couldn't deduce the destination file.
If you think it is not the right way to go, I can lose the printing altogether but I think it doesn't harm to print it only if we have the information.